src/main/java/org/spongepowered/api/scoreboard/Objective.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.scoreboard;
+
+import com.google.common.base.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * An objective on a scoreboard that can show scores specific to entries. This
+ * objective is only relevant to the display of the associated {@link
+ * #getScoreboard() scoreboard}.
+ */
+public interface Objective {
+
+    /**
+     * Gets the name of this Objective.
+     *
+     * @return The objective's name
+     */
+    String getName();
+
+    /**
+     * Gets the name displayed to players.
+     *
+     * @return The objective's display name
+     */
+    String getDisplayName();
+
+    /**
+     * Sets the name displayed to players.
+     *
+     * @param displayName Display name to set
+     * @throws IllegalArgumentException if displayName is longer than 32
+     *     characters.
+     */
+    void setDisplayName(String displayName) throws IllegalArgumentException;
+
+    /**
+     * Gets the criteria this objective tracks.
+     *
+     * @return this objective's criteria
+     */
+    String getCriteria();
+
+    /**
+     * Gets whether objective's scores can be modified directly by a plugin.
+     *
+     * @return true if scores are modifiable
+     * @see Criteria#HEALTH
+     */
+    boolean isModifiable();
+
+    /**
+     * Gets the scoreboard to which this objective is attached, if present.
+     *
+     * @return Owning scoreboard, if present
+     */
+    Optional<Scoreboard> getScoreboard();
+
+    /**
+     * Sets the scoreboard to which this objective is attached.
+     *
+     * @param scoreboard The scoreboard
+     */
+    void setScoreboard(Scoreboard scoreboard);
+
+    /**
+     * Sets this objective to display on the specified slot for the
+     * scoreboard, removing it from any other display slot.
+     *
+     * If another objective is set to the same display slot, that objective will
+     * have it's display slot set to <code>null</code>.
+     *
+     * @param slot display slot to change, or null to not display
+     */
+    void setDisplaySlot(@Nullable DisplaySlot slot);
+
+    /**
+     * Gets the display slot this objective is displayed at.
+     *
+     * @return the display slot for this objective, if set
+     */
+    Optional<DisplaySlot> getDisplaySlot();
+
+    /**
+     * Gets an entry's Score for this Objective.
+     *
+     * @param entry Entry for the Score
+     * @return Score tracking entry specified
+     */
+    Score getScore(String entry);
```

Comment:
It would be useful to duplicate the setScore and getScore methods from Score here in case a developer doesn't need an actual Score instance(which is most cases)